src/appleseed/renderer/kernel/lighting/tracer.h

```diff
@@ -328,11 +332,14 @@ inline double Tracer::trace_between(
 {
     if (m_assume_no_alpha_mapping)
     {
+        const foundation::Vector3d direction = target - origin;
+        const double dist = foundation::norm(direction);
+
         const ShadingRay ray(
             origin,
-            target - origin,
+            direction / dist,
             0.0,                        // ray tmin
-            1.0 - 1.0e-6,               // ray tmax
+            dist - 1.0e-6,              // ray tmax
```

Comment:
There's a subtlely here: while `1.0 - 1.0e-6` was guaranteed to evaluate to less than 1.0, it might not be the case of `dist - 1.0e-6` due to floating point cancellation. While still less than idea, `dist * (1.0 - 1.0e-6)` is probably better.